### PR TITLE
Add editor

### PR DESCRIPTION
**Pull request by dspencer001**

### CODE REVIEW

File: client/config/settings.js [hunk 1]
```diff
@@ -54,7 +54,8 @@ module.exports = {
 
   cssEntries: {
     player_styles : `${clientAppPath}styles/player_styles.js`,
-    author_styles : `${clientAppPath}styles/author_styles.js`
+    author_styles : `${clientAppPath}styles/author_styles.js`,
+    editor_styles : `${clientAppPath}styles/editor_styles.js`
```
Comment (BenHSmith13) [inline client/config/settings.js]: What is editor styles, is this for another app?

File: client/js/_author/components/common/oea_editor.jsx [hunk 2]
```diff
@@ -0,0 +1,41 @@
+import React       from 'react';
+import TinyWrapper from './tiny_wrapper';
+
+export default class OeaEditor extends React.Component{
+  static propTypes = {};
```
Comment (BenHSmith13) [inline client/js/_author/components/common/oea_editor.jsx]: yeah

File: client/js/_author/components/common/tiny_wrapper.jsx [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+  constructor() {
+    super();
+
+    this.state = {
+      imageCallback: null,
+      imageGuid: null,
+      uploadingImage: false
+    };
+
+    this.id = guid();
```
Comment (BenHSmith13) [inline client/js/_author/components/common/tiny_wrapper.jsx]: I'm a loser baby, so why don't you kill me

File: client/js/_author/components/common/tiny_wrapper.jsx [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+function select(state) {
+  return {
+    uploadedAssets: state.uploadedAssets,
+    apiUrl: state.settings.api_url,
+  };
+}
+
+export class TinyWrapper extends React.Component {
+  static propTypes = {
+    text: React.PropTypes.string,
```
Comment (BenHSmith13) [inline client/js/_author/components/common/tiny_wrapper.jsx]: you are missing some, turn your linter back on

File: client/js/_author/components/common/tiny_wrapper.jsx [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+  shouldComponentUpdate() {
+    // TinyMCE doesn't handle rerendering very well, and everything dealing
+    // with the content inside of it is completely managed by tinyMCE, so we
+    // never rerender. All state that needs to be updated is done so in
+    // componentWillReceiveProps or in callbacks from tinyMCE.
+    return false;
+  }
+
+  uploadImage(e) {
+    const imageGuid = guid();
```
Comment (BenHSmith13) [inline client/js/_author/components/common/tiny_wrapper.jsx]: check that there is actually a file

File: client/js/_author/components/common/tiny_wrapper.jsx [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+      paste_data_images: true
+    };
+  }
+
+  render() {
+    return (
+      <div>
+        <div id={this.id} />
+        <TinyMCE
+          className="editable_component"
```
Comment (BenHSmith13) [inline client/js/_author/components/common/tiny_wrapper.jsx]: kill this class name

File: client/js/_author/components/common/tiny_wrapper.jsx [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+      imageGuid,
+      this.props.bankId,
+      this.props.apiUrl
+    );
+  }
+
+  tinyMCEConfig() {
+    // Whenever you add a plugin, make sure that it is imported above.
+    return {
+      selector: ".editable_component",
```
Comment (BenHSmith13) [inline client/js/_author/components/common/tiny_wrapper.jsx]: dont need this

File: client/js/_author/components/common/tiny_wrapper.jsx [hunk 8]
```diff
@@ -0,0 +1,113 @@
...
+    };
+  }
+
+  render() {
+    return (
+      <div>
+        <div id={this.id} />
+        <TinyMCE
+          className="editable_component"
+          content={this.props.text || "Question Text"}
```
Comment (BenHSmith13) [inline client/js/_author/components/common/tiny_wrapper.jsx]: placeholder text or nothing

File: client/js/middleware/qbank.js [hunk 9]
```diff
@@ -250,6 +251,11 @@ const qbank = {
     );
   },
 
+  [AssetConstants.UPLOAD_IMAGE]: {
+    method : Network.POST,
+    url    : action => `https://qbank-clix-dev.mit.edu/api/v1/repository/repositories/${action.bankId}/assets`,
```
Comment (BenHSmith13) [inline client/js/middleware/qbank.js]: will need to use the url from settings

File: client/js/_author/components/common/tiny_wrapper.jsx (hunk 5 above)
Comment (dspencer001) [inline client/js/_author/components/common/tiny_wrapper.jsx]: Turns out this is unnecessary, I thought I'd seen this callback get called without a file but I think that's impossible

File: client/js/_author/components/common/tiny_wrapper.jsx (hunk 3 above)
Comment (dspencer001) [inline client/js/_author/components/common/tiny_wrapper.jsx]: Added this back in, tinymce breaks because of the dots in the qbank ids...

Comment (BenHSmith13) [inline client/js/_author/components/common/tiny_wrapper.jsx]: Awesome

**Comment by BenHSmith13** [comment]
# LGTM
!(Looks Good)[http://image.slidesharecdn.com/ndsmeetup8-151114064348-lva1-app6891/95/javascript-jquery-ndsmeetup8-31-638.jpg?cb=1447565972]

File: client/js/_author/components/common/tiny_wrapper.jsx [hunk 10]
```diff
@@ -100,6 +100,7 @@ export class TinyWrapper extends React.Component {
           onFocus={this.props.onFocus}
         />
         <input
+          style={{width: '0.1px', height: '0.1px', display: 'none'}}
```
Comment (BenHSmith13) [inline client/js/_author/components/common/tiny_wrapper.jsx]: inline style? written by David Spencer? I am shocked!

Comment (dspencer001) [inline client/js/_author/components/common/tiny_wrapper.jsx]: Brandon will style it later lol

File: client/js/constants/genus_types.js [hunk 11]
```diff
@@ -24,14 +24,7 @@ export const types = {
...
 export function getQbankType(type) {
-  switch (type) {
-    case types.item.multipleChoice:
-      return 'multipleChoice';
-
-    case types.default:
-    default:
-      return null;
-  }
+  return _.findKey(types.item, (genusType) => type === genusType) || null;
```
Comment (BenHSmith13) [inline client/js/constants/genus_types.js]: will this not return null?

Comment (dspencer001) [inline client/js/constants/genus_types.js]: it returns undefined

Comment (dspencer001) [inline client/js/constants/genus_types.js]: which may be ok, I just figured I'd leave it returning what you had the switch returning

Comment (BenHSmith13) [inline client/js/constants/genus_types.js]: fair enough

**Pull request merged**